src/core/cache-storage.ts

```diff
@@ -39,20 +39,15 @@ export class Cache {
 
     constructor(private readonly context: Context, private readonly _options: ResourceOptions) {}
 
-    addImage(src: string): Promise<void> {
-        const result = Promise.resolve();
-        if (this.has(src)) {
-            return result;
-        }
-
+    addImage(src: string): boolean {
```

Comment:
The `boolean` value is never used, should it be `void`?

Reply:
True, it's never used, but I decided to extend this function, because there is a case when an image won't be added to the cache and false is returned in this case.